bench/workgen/workgen.cxx

```diff
@@ -248,6 +281,231 @@ WorkloadRunner::start_table_idle_cycle(WT_CONNECTION *conn)
+             */
+            ASSERT(std::string(e.what()).find("No such file or directory") != std::string::npos);
+            continue;
+        }
+    }
+    return result / (WT_THOUSAND * WT_THOUSAND);
+}
+
+/*
+ * This function will create "dynamic_create_count" tables every "dynamic_create_period" seconds. It
+ * will also monitor the database size and stop creating tables once the size crosses
+ * "dynamic_create_target". It will restart table creation if the database size drops below
+ * "dynamic_create_trigger".
+ */
+int
+WorkloadRunner::start_dynamic_create(WT_CONNECTION *conn)
+{
+    WT_SESSION *session;
+    if (conn->open_session(conn, nullptr, nullptr, &session) != 0) {
+        THROW("Error opening a session.");
+    }
+
+    WT_DECL_RET;
+    ContextInternal *icontext = _workload->_context->_internal;
+    char uri[BUF_SIZE];
+    uint64_t total_creates = 0;
+    uint32_t db_size = get_dir_size_mb(_wt_home);
+
+    // Initially we start creating tables if the database size is less than the create target.
+    bool creating = db_size < _workload->options.dynamic_create_target;
+
+    while (!stopping) {
+        /*
+         * If we have been creating tables, continue until we reach the create target size. If we
+         * have not been creating tables, begin to do so if the database size falls below the create
+         * trigger.
+         */
+        db_size = get_dir_size_mb(_wt_home);
+        if (creating) {
+            creating = db_size < _workload->options.dynamic_create_target;
+            if (!creating) {
+                VERBOSE(*_workload,
+                  "Stopped creating new tables. db_size now "
+                    << db_size << " MB has reached the create target of "
+                    << _workload->options.dynamic_create_target << " MB.");
+            }
+        } else {
+            creating = db_size < _workload->options.dynamic_create_trigger;
+            if (creating) {
+                VERBOSE(*_workload,
+                  "Started creating new tables. db_size now "
+                    << db_size << " MB has reached the create trigger of "
+                    << _workload->options.dynamic_create_trigger << " MB.");
+            }
+        }
+        if (!creating) {
+            sleep(_workload->options.dynamic_create_period);
+            continue;
+        }
+
+        for (int creates = 0; creates < _workload->options.dynamic_create_count;
+             ++creates, ++total_creates) {
+            snprintf(uri, BUF_SIZE, "table:test_dynamic_%" PRIu64, total_creates);
+
+            // Make sure that this table is not a part of static table set, continue if it is.
+            if (icontext->_tint.count(uri) > 0 || icontext->_dyn_tint.count(uri) > 0)
+                continue;
+
+            // Create the table.
+            if ((ret = session->create(session, uri, "key_format=S,value_format=S")) != 0) {
+                if (ret == EBUSY)
+                    continue;
```

Comment:
ditto, this will trigger the next iteration and increment the counters while no work has been done.

Reply:
This is intentional for now. I will change this when I generate random tables and not sequential ones.